### DISCUSSION TITLE
How to get task id

### DISCUSSION THREAD

**Discussion by shb256** [semaphoreui/semaphore#2275] [Q&A] [answered]
Hello,

I trigger semaphore with a webhook unfortunately i am not able to use the response.
(How) Can I access the task id in the playbook? ist there a variable wich i can pass to ansible?

thanks

**Comment by shb256** [comment]
Is there really no posibility to get the Task  ID as a variable inside the playbook?

**Reply by SirYaro** [reply]
I'm looking for the similar feature as well.

**Reply by sebaboben** [reply, marked as answer]
When running a playbook with the verbose tag I found this line:

extra_vars: ('{"semaphore_vars":{"task_details":{"id":308,"url":null,"username":"admin"}}}', 'rbose')

This shows some useful variables such as the task ID. So if you need to fetch it from it's respective playbook run you just need to use a task like this one.

    - name: Fetch Task ID
      set_fact:
        taskid: "{{ semaphore_vars.task_details.id }}"
